sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -313,13 +313,19 @@ functionCall
     ;
 
 nestedFunction
-    : NESTED LR_BRACKET nestedField RR_BRACKET
+    : NESTED LR_BRACKET
+    (nestedField | nestedField COMMA nestedPath | nestedPath COMMA expression)
+    RR_BRACKET
     ;
 
 nestedField
     : ID DOT ID (DOT ID)*
     ;
 
+nestedPath
+    : ID (DOT ID)*
+    ;
```

Comment:
Should we support backticks or single/double quotes? A field could have dashes and spaces in the name.